// styled-modal-component/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Modal", {
  enumerable: true,
  get: function get() {
    return _Modal.Modal;
  }
});
Object.defineProperty(exports, "ModalWrapper", {
  enumerable: true,
  get: function get() {
    return _ModalWrapper.ModalWrapper;
  }
});
Object.defineProperty(exports, "ModalBody", {
  enumerable: true,
  get: function get() {
    return _ModalBody.ModalBody;
  }
});
Object.defineProperty(exports, "ModalContent", {
  enumerable: true,
  get: function get() {
    return _ModalContent.ModalContent;
  }
});
Object.defineProperty(exports, "ModalDialog", {
  enumerable: true,
  get: function get() {
    return _ModalDialog.ModalDialog;
  }
});
Object.defineProperty(exports, "ModalFooter", {
  enumerable: true,
  get: function get() {
    return _ModalFooter.ModalFooter;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function get() {
    return _ModalHeader.ModalHeader;
  }
});
Object.defineProperty(exports, "ModalTitle", {
  enumerable: true,
  get: function get() {
    return _ModalTitle.ModalTitle;
  }
});
Object.defineProperty(exports, "ModalBackdrop", {
  enumerable: true,
  get: function get() {
    return _ModalBackdrop.ModalBackdrop;
  }
});

var _Modal = require("./components/Modal");

var _ModalWrapper = require("./components/ModalWrapper");

var _ModalBody = require("./components/ModalBody");

var _ModalContent = require("./components/ModalContent");

var _ModalDialog = require("./components/ModalDialog");

var _ModalFooter = require("./components/ModalFooter");

var _ModalHeader = require("./components/ModalHeader");

var _ModalTitle = require("./components/ModalTitle");

var _ModalBackdrop = require("./components/ModalBackdrop");
